ledger/src/blocktree.rs

```diff
@@ -252,10 +253,10 @@ impl Blocktree {
         }))
     }
 
-    pub fn slot_data_iterator(
-        &self,
+    pub fn slot_data_iterator<'a>(
```

Comment:
why is the explicit lifetime necessary now?

Reply:
Otherwise the returned trait conservatively gets pegged to `'static` https://doc.rust-lang.org/reference/lifetime-elision.html#default-trait-object-lifetimes